### PR TITLE
[CRITICAL] Adjust <WorkspaceWorkflowsApprovalsApproverPage /> for Approver Selection

### PR DESCRIPTION
**Pull request by blazejkustra**
### Details

[Link to the doc](https://docs.google.com/document/d/12X-hsNo9BBRcG5NSS1JzyjWD1w-iM0uciQJSD1i5mZc/edit). 

#### UI changes

image

`WorkspaceWorkflowsApprovalsApproverPage` is split into two components:
- `WorkspaceWorkflowsApprovalsApproverPageNew` - The new refreshed UI with logic changes, this variant is displayed only when `canUseWorkflowsAdvancedApproval` beta is turned on + `approverIndex` screen param has to be provided.
- `WorkspaceWorkflowsApprovalsApproverPageOld` - The old screen left for backward compatibility, it will be displayed for the old flow:
image

#### Onyx actions:

Clicking on an approver will set the approver with `setApprovalWorkflowApprover` action (more info in the [Onyx setup section](https://docs.google.com/document/d/12X-hsNo9BBRcG5NSS1JzyjWD1w-iM0uciQJSD1i5mZc/edit#heading=h.pdxemoaeemuw)) and navigate to the ‘Create approval’ page (`‘settings/workspaces/:policyID/workflows/approvals/new’`) or ‘Edit approval’ page (`‘settings/workspaces/:policyID/workflows/approvals/:firstApproverEmail/edit’`) based on `isBeingEdited` property in Onyx.

### Fixed Issues
$ https://github.com/Expensify/App/issues/45955
PROPOSAL: N/A

### Tests
- [x] Open Workflows page (it has to be turned on in `More features` settings page)
- [x] Locally edit `canUseWorkflowsAdvancedApproval` function to return true (or use `canUseWorkflowsAdvancedApproval` beta)
- [x] Click on the 'Add approval workflow' button, it should navigate to 'Expenses from' page, select some users and go to the next step
- [x] Verify that Approver page is displayed
  - "Find a member" input should filter users
  - You can select and deselect an approver
- [x] Next button should navigate to the create new workflow page

### Offline tests
N/A

### QA Steps
Same as tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[android.webm](https://github.com/user-attachments/assets/0b53af1e-f9c1-4829-92ad-9055d1eabf32)

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[android-web.webm](https://github.com/user-attachments/assets/58bd9a23-4644-45b7-953e-131cc3f02c6f)

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/182fd2ef-c6a6-4a9b-8bc3-af7f4ce7591d

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/0f6e275a-ca9f-4c0c-a832-7605545e490d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/458fb44b-c119-4c0c-a806-506940392b56

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/e0b43812-93df-4bff-be69-9128265dbf38

</details>

### CODE REVIEW

**Comment by blazejkustra** [comment]
To push things forward I opened the PR without videos, I'm sick and will add them later. @suneox please have a look at the PR 😄

**Comment by blazejkustra** [comment]
> > To push things forward I opened the PR without videos, I'm sick and will add them later. @suneox please have a look at the PR 😄
> 
> @blazejkustra Sure, how can I access the document to get more context? I’ve requested access via email `ptcong411@gmail.com`.

I shared it with you 🚀

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsApproverPage.tsx
Comment (blazejkustra) [reply src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsApproverPage.tsx:87]: For creation flow we could use `approverEmail` instead of `approverIndex`, however when editing an existing approval workflow (which will use the same approver page) we need an information which approver is currently edited, sole `approverEmail` would not be enough as there might be a rare ocasion where there are duplicated emails in one workflow.

Fixed in https://github.com/Expensify/App/pull/46510/commits/6d6a4a67df7fa71bdc51df16a3d4e2f8e11ee81c. My logic is simple:
- When selecting an approver user have to click on 'Next' button to set the approver in the onyx
  - Creation flow: When going back to "expenses from" screen, the selected approver will be cleared
  - Edition flow: When going back to "edit approval workflow" screen, the selected approver won't be cleared

Comment (blazejkustra) [reply src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsApproverPage.tsx:87]: Let me know what do you think @suneox 
https://github.com/user-attachments/assets/af1e3953-3e9b-4318-ba5b-2fe09639d523

**Review by suneox — APPROVED** [review]
Works as expected on all platforms, and the change set LGTM

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsApproverPage.tsx
Comment (tgolen) [inline src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsApproverPage.tsx:74]: Would you mind changing the name of this component to `WorkspaceWorkflowsApprovalsApproverPageBeta` and I would just keep the name of the "old" component the way it was before. The only reason is that "new" and "old" hold no meaning once this PR is merged.

**Review by tgolen — CHANGES_REQUESTED** [review]

**Review by tgolen — APPROVED** [review]
Thanks!

**Comment by tgolen** [comment]
@suneox Did you ever get paid for your C+ review of this PR? If not, I think we need to handle that in https://github.com/Expensify/App/issues/45956 before the issue can be closed out.

**Pull request merged**
